### PR TITLE
Add modification_time into system.detached_parts

### PR DESCRIPTION
**Pull request by azat**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add modification_time into system.detached_parts

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
This is an automated comment for commit 1b5b4b76e8a45e0ce50578d6c958670db818ea3c with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/54506/1b5b4b76e8a45e0ce50578d6c958670db818ea3c/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Storages/System/StorageSystemDetachedParts.cpp [hunk 1]
```diff
@@ -230,6 +231,19 @@ class DetachedPartsSource : public ISource
                 size_t bytes_on_disk = parts_sizes.at(p_id - begin).load();
                 new_columns[res_index++]->insert(bytes_on_disk);
             }
+            if (columns_mask[src_index++])
+            {
+                Poco::Timestamp modification_time;
```
Comment (alexey-milovidov) [inline src/Storages/System/StorageSystemDetachedParts.cpp:236]: Looks suspicious - initialize to zero.

Comment (azat) [reply src/Storages/System/StorageSystemDetachedParts.cpp]: Actually the default ctor of the Timestamp uses "now", and that was the intention

And your change (`s/Poco::Timestamp modification_time;/Poco::Timestamp modification_time{};/`) did not change anything, there will still be "now" if `stat` fails.

Comment (alexey-milovidov) [reply src/Storages/System/StorageSystemDetachedParts.cpp]: Thank you! I was only reading the diff, and the code was suspicious for me.

**Pull request merged**